tests/ClientBuilderTest.php

```diff
@@ -290,6 +290,22 @@ public function testCreateWithNoOptionsIsTheSameAsDefaultOptions(): void
             ClientBuilder::create([])
         );
     }
+
+    public function testCreateWithHttpProxyAndCustomTransportThrowsException(): void
+    {
+        $options = new Options([
+            'dsn' => 'http://public:secret@example.com/sentry/1',
+            'http_proxy' => 'some-proxy',
+        ]);
+
+        $clientBuilder = new ClientBuilder($options);
+        $clientBuilder->setHttpClient($this->createMock(HttpAsyncClient::class));
+
+        $this->expectException(\RuntimeException::class);
+        $this->expectExceptionMessage('http_proxy');
```

Comment:
I don't know how this test can pass since the message is different from the one in the codebase. I believe that the reason is that this method just asserts that the string is contained in the message, but since it's not clear please use the full message